tests/e2e/scenarios/test_reborn_webui_v2_custom_mcp.py

```
@@ -0,0 +1,267 @@
+            if package_id == "custom-weather-mcp":
+                await fulfill(route, _setup_projection(auth_kind))
+                return
+        if path.endswith("/setup") and request.method == "POST":
+            package_id = unquote(path.removeprefix("/api/webchat/v2/extensions/").removesuffix("/setup"))
+            if package_id == "custom-weather-mcp":
+                body = json.loads(request.post_data or "{}")
+                setup_submissions.append(body)
+                if setup_result == "wrong":
+                    await fulfill(route, {"success": False, "message": "Token was rejected"})
+                    return
+                installed[:] = [_registered_extension(auth_kind, "active")]
```

Ruff (0.16.0)

[error] Undefined name `_registered_extension`

(F821)